lib/segment/src/index/hnsw_index/search_context.rs

```diff
@@ -24,19 +22,13 @@ impl SearchContext {
     }
 
     pub fn lower_bound(&self) -> ScoreType {
-        match self.nearest.top() {
-            None => ScoreType::min_value(),
-            Some(worst_of_the_best) => worst_of_the_best.score,
-        }
+        self.nearest.threshold()
     }
 
     /// Updates search context with new scored point.
     /// If it is closer than existing - also add it to candidates for further search
     pub fn process_candidate(&mut self, score_point: ScoredPointOffset) {
-        let was_added = match self.nearest.push(score_point) {
-            None => true,
-            Some(removed) => removed.idx != score_point.idx,
-        };
+        let was_added = self.nearest.push(score_point);
```

Comment:
why is the deduplication based on `removed.idx != score_point.idx,` not required anymore?

Reply:
We can't return index of the point if it was discarded or not, because we would need to return a whole list in this case. So the TopK return type is different and it is just bool